File: certificados-client/src/service/DegreeCatalogService.js
import DegreeCatalogApi from "../api/DegreeCatalogApi";

import Alerts from "../shared/alerts";

export default {
  getAntecedents: async () => {
    const response = await DegreeCatalogApi.getAntecedents();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  getReasons: async () => {
    const response = await DegreeCatalogApi.getReasons();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  getAuths: async () => {
    const response = await DegreeCatalogApi.getAuths();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  getModalities: async () => {
    const response = await DegreeCatalogApi.getModalities();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  getSigners: async () => {
    const response = await DegreeCatalogApi.getSigners();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  getSocialService: async () => {
    const response = await DegreeCatalogApi.getSocialService();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  getStates: async () => {
    const response = await DegreeCatalogApi.getStates();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  getCarrer: async (curp) => {
    const response = await DegreeCatalogApi.getCarrer(curp);
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  getSchools: async (stateId) => {
    const response = await DegreeCatalogApi.getSchools(stateId);
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  }, 
  getCarrers: async (schoolId) => {
    const response = await DegreeCatalogApi.getCarrers(schoolId);
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  }, 
  getAllStates: async () => {
    const response = await DegreeCatalogApi.getAllStates();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  schoolsNormalAll: async () => {
    const response = await DegreeCatalogApi.schoolsNormalAll();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  careerAllDgp: async () => {
    const response = await DegreeCatalogApi.careerAllDgp();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  searSchoolDgp: async (schoolId) => {
    const response = await DegreeCatalogApi.searSchoolDgp(schoolId);
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  studentPeriodDate: async (curp) => {
    const response = await DegreeCatalogApi.studentPeriodDate(curp);
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
  stateListAll: async () => {
    const response = await DegreeCatalogApi.stateListAll();
    if (!response.success) Alerts.error("Ha ocurrido un error", response.message);
    return response;
  },
};

File: certificados-server/src/main/java/mx/edu/cecyte/sisec/service/degree/DegreeCatalogService.java
package mx.edu.cecyte.sisec.service.degree;

import mx.edu.cecyte.sisec.classes.degree.StudentPeriodDate;
import mx.edu.cecyte.sisec.dto.catalogs.Catalog;
import mx.edu.cecyte.sisec.dto.degree.DgpSchoolSelect;
import mx.edu.cecyte.sisec.model.student.Student;
import mx.edu.cecyte.sisec.model.users.User;
import mx.edu.cecyte.sisec.queries.UserQueries;
import mx.edu.cecyte.sisec.queries.degree.DegreeCatalogQueries;
import mx.edu.cecyte.sisec.shared.AppCatalogs;
import mx.edu.cecyte.sisec.shared.UserSession;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Set;

@Service
public class DegreeCatalogService {
    @Autowired private UserQueries userQueries;
    @Autowired private DegreeCatalogQueries catalogQueries;

    public List<Catalog> getAntecedents() {
        return catalogQueries.getAntecedents();
    }

    public List<Catalog> getReasons() {
        return catalogQueries.getReasons();
    }

    public List<Catalog> getAuths() {
        return catalogQueries.getAuths();
    }

    public List<Catalog> getModalities() {
        return catalogQueries.getModalities();
    }

    public List<Catalog> getSigners() {
        return catalogQueries.getSigners();
    }

    public List<Catalog> getSocialService() {
        return catalogQueries.getSocialService();
    }

    public List<Catalog> getDegreeStates(String username) {
        User user = userQueries.getUserByUsername(username);
        Set<Integer> availableStateIds = userQueries.getAvailableStateIdsByAdminUser(user);
        return catalogQueries.getDegreeStates(availableStateIds);
    }

    public List<Catalog> getDegreeCarrer(String curp) {
        Integer schoolId = catalogQueries.getchooltId(curp);
        return catalogQueries.getDegreeCarrer(schoolId);
    }

    public List<DgpSchoolSelect> getDegreeSchools(Integer stateId, String username) {
        User user = userQueries.getUserByUsername(username);
        Set<Integer> availableSchoolIds = userQueries.getAvailableSchoolIdsByAdminUserV2(stateId,user, AppCatalogs.isState);
        return catalogQueries.getDegreeSchools(stateId, availableSchoolIds);
    }

    public List<Catalog> getDegreeCarrers(Integer schoolId) { return catalogQueries.getDegreeCarrers(schoolId); }

    public List<Catalog> degreeAllStates(String username) {
        User user = userQueries.getUserByUsername(username);

        Set<Integer> availableStateIds = userQueries.getAvailableStateIdsByAdminUser(user);
        return catalogQueries.degreeAllStates(availableStateIds);
    }

    public List<Catalog> careerAllDgp() {
        return catalogQueries.careerAllDgp();
    }

    public List<Catalog> schoolsNormalAll() {
        return catalogQueries.schoolsNormalAll();
    }

    public List<Catalog> searSchoolDgpFindById(Integer schoolId) {
        return catalogQueries.searSchoolDgpFindById(schoolId);
    }

    public List<StudentPeriodDate> studentPeriodDate(String curp) {
        User user = userQueries.getUserByUsername(curp);
        return catalogQueries.studentPeriodDate(user.getStudent().getId());
    }

    public List<Catalog> stateListAll(UserSession userSession) {
        return catalogQueries.stateListAll();
    }
}
